functions/venue.js

```diff
@@ -372,14 +372,14 @@ const createBaseUpdateVenueData = (data, doc) => {
     updated.mapBackgroundImageUrl = data.mapBackgroundImageUrl;
   }
 
-  if (data.parentId) {
-    updated.parentId = data.parentId;
-  }
-
   if (data.roomVisibility) {
     updated.roomVisibility = data.roomVisibility;
   }
 
+  if (typeof data.parentId === "string") {
+    updated.parentId = data.parentId;
+  }
+
```

Comment:
No check for self reference on input causes bug.
![sparkle-admin-parent-id-01](https://user-images.githubusercontent.com/79229621/135251329-c4923992-b412-468a-950d-8e73ee49d65d.png)